apps/server/src/modules/authorization-rules/rules/instance.rule.ts

```diff
@@ -20,21 +20,36 @@ export class InstanceRule implements Rule<Instance> {
 	}
 
 	public isApplicable(user: User, object: unknown): boolean {
-		const isMatched = object instanceof Instance;
+		const isApplicable = object instanceof Instance;
 
-		return isMatched;
+		return isApplicable;
 	}
 
 	public hasPermission(user: User, entity: Instance, context: AuthorizationContext): boolean {
-		const hasPermission = this.authorizationHelper.hasAllPermissions(user, context.requiredPermissions);
+		let hasPermission = false;
 
-		// As temporary solution until the user with write access to instance added as group, we must check the role.
+		if (context.action === Action.read) {
+			hasPermission = this.hasReadAccess(user, context);
+		}
 		if (context.action === Action.write) {
-			const hasRole = this.authorizationHelper.hasRole(user, RoleName.SUPERHERO);
-
-			return hasPermission && hasRole;
+			hasPermission = this.hasWriteAccess(user, context);
 		}
 
 		return hasPermission;
 	}
+
+	private hasReadAccess(user: User, context: AuthorizationContext): boolean {
+		const hasPermission = this.authorizationHelper.hasAllPermissions(user, [
+			Permission.INSTANCE_VIEW,
+			...context.requiredPermissions,
+		]);
+
+		return hasPermission;
+	}
+
+	private hasWriteAccess(user: User, context: AuthorizationContext): boolean {
+		const hasPermission = false;
```

Comment:
Ein Kommentar ala "Is currently not implemented. No user data for this context exists in database." wäre noch gut.